src/main/java/io/cdap/plugin/gcp/speech/SpeechToTextTransform.java

```diff
@@ -68,15 +75,16 @@
   @Override
   public void configurePipeline(PipelineConfigurer configurer) throws IllegalArgumentException {
     super.configurePipeline(configurer);
-    Schema inputSchema = configurer.getStageConfigurer().getInputSchema();
-    config.validate(inputSchema);
+    StageConfigurer stageConfigurer = configurer.getStageConfigurer();
+    Schema inputSchema = stageConfigurer.getInputSchema();
+    config.validate(inputSchema, stageConfigurer);
     configurer.getStageConfigurer().setOutputSchema(getSchema(inputSchema));
   }
 
   @Override
   public void prepareRun(StageSubmitterContext context) throws Exception {
     super.prepareRun(context);
-    config.validate(context.getInputSchema());
+    config.validate(context.getInputSchema(), context.getFailureCollector());
```

Comment:
it seems a bit odd that here we do getFailureCollector(), but at configure time the StageConfigurer is a failure collector. What do you think about modifying the API such that PipelineConfigurer has a getFailureCollector() method?

Reply:
yea makes sense.